src/scores/categorical/risk_matrix_impl.py

```diff
@@ -0,0 +1,498 @@
+        severity_dim: the dimension specifying severity categories.
+        prob_threshold_dim: the dimension in ``decision_weights`` specifying probability thresholds.
+        threshold_assignment: Either "upper" or "lower". Specifies whether the probability
+            intervals defining the certainty categories, with endpoints given by the
+            decision thresholds in ``decision_weights``, are left or right closed. That is,
+            whether the probability decision threshold is included in the upper (left closed)
+            or lower (right closed) certainty category. Defaults to "lower".
+        reduce_dims: Optionally specify which dimensions to reduce when calculating the
+            mean risk matrix score, where the mean is taken over all forecast cases.
+            All other dimensions will be preserved. As a special case, 'all' will allow
+            all dimensions to be reduced. Only one of ``reduce_dims`` and ``preserve_dims``
+            can be supplied. The default behaviour if neither are supplied is to reduce all dims.
+        preserve_dims: Optionally specify which dimensions to preserve when calculating the
+            mean risk matrix score, where the mean is taken over all forecast cases.
+            All other dimensions will be reduced. As a special case, 'all' will allow
+            all dimensions to be preserved, apart from ``severity_dim`` and ``prob_threshold_dim``.
+            Only one of ``reduce_dims`` and ``preserve_dims`` can be supplied. The default
+            behaviour if neither are supplied is to reduce all dims.
+        weights: Optionally provide an array for weighted averaging (e.g. by area, by latitude,
+            by population, custom) of scores across all forecast cases.
+
+    Returns:
+        An xarray object of risk matrix scores, averaged according to specified weights
+        across appropriate dimensions.
+
+    Raises:
+        ValueError: if ``severity_dim`` is not a dimension of ``fcst``, ``obs`` or ``decision_weights``.
+        ValueError: if ``severity_dim`` is a dimension of ``weights``.
+        ValueError: if ``prob_threshold_dim`` is not a dimension of ``decision_weights``.
+        ValueError: if ``prob_threshold_dim`` is a dimension of ``weights``.
+        ValueError: if ``decision_weights`` does not have exactly 2 dimensions.
+        ValueError: if ``fcst`` values are negative or greater than 1.
+        ValueError: if ``obs`` values are other than 0, 1 or nan.
+        ValueError: if ``prob_threshold_dim`` coordinates are not strictly between 0 and 1.
+        ValueError: if ``severity_dim`` coordinates differ for any of ``fcst``, ``obs`` or ``decision_weights``.
+        ValueError: if ``threshold_assignment`` is not "upper" or lower".
+
+    References:
+        - Taggart, R. J., & Wilke, D. J. (2025). Warnings based on risk matrices: a coherent framework
+          with consistent evaluation. Nat. Hazards Earth Syst. Sci., 25, 2657–2677.
```

Comment:
```suggestion
          with consistent evaluation. Natural Hazards and Earth System Sciences, 25(8), 2657–2677.
```
Suggested updates to follow [APA 7 style guide for journal article references](https://apastyle.apa.org/style-grammar-guidelines/references/examples/journal-article-references):
- Full name of journal (not abbreviations)
- Add issue number